### PR TITLE
[TS SDK] Add account fetch queries

### PR DESCRIPTION
**Pull request by 0xmaayan**
### Description

This PR adds the account fetch queries
- `getData, getModules, getModule, getTransactions, getResources, getResource`
It creates an account namespace in the main Aptos class that holds the account fetch queries namespace. Each namespace function calls an internal function that actually does the query. this is so other namespaces and processes can access these methods without depending on the entire account namespace and without having a dependency cycle error.

In addition, this PR
- generates openAPI response types. While we use some custom types, we still need the generated types like `Transaction, MoveFunction, MoveStruct` - openAPI response types will be completely removed once API v2 is ready.
- creates a `getRequestUrl` function and `AptosApiTypes` to easily interact with the needed service to query from. This helps preventing using, for example `this.fullnode!` when executing a query, since `fullnode` might be undefined.
- adds initial account queries tests with a TODO to add more tests once sdk v2 supports faucet and transaction operations.
- handle some lint errors from previous PRs

### CODE REVIEW

File: ecosystem/typescript/sdk_v2/src/types/generated/models/MoveStructValue.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+     * For example:
+     * `{ "created": "0xa550c18", "role_id": "0" }`
+     *
+     * **Special serialization for Move stdlib types**:
+     * - [0x1::string::String](https://github.com/aptos-labs/aptos-core/blob/main/language/move-stdlib/docs/ascii.md)
+     * is serialized into `string`. For example, struct value `0x1::string::String{bytes: b"Hello World!"}`
+     * is serialized as `"Hello World!"` in JSON.
+     *
+     */
+    export type MoveStructValue = {
```
Comment (xbtmatt) [inline ecosystem/typescript/sdk_v2/src/types/generated/models/MoveStructValue.ts:47]: Is this correct? What's the point of this if so haha 🤔

File: ecosystem/typescript/sdk_v2/src/types/generated/models/MoveStructValue.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    };
```
Comment (xbtmatt) [inline ecosystem/typescript/sdk_v2/src/types/generated/models/MoveStructValue.ts:48]: Is this correct? What's the point of this if so haha 🤔

**Comment by xbtmatt** [comment]
Is this file: ecosystem/typescript/sdk_v2/src/internal/account.ts

the main one with the changes? seems like it has all of the `getData, getModules, getModule, getTransactions, getResources, getResource` functions

The rest is mostly generated types, right? Just making sure which part is actually new vs old generated stuff that we will phase out

File: ecosystem/typescript/sdk_v2/src/types/generated/models/MoveStructValue.ts (hunk 2 above)
Comment (0xmaayan) [reply ecosystem/typescript/sdk_v2/src/types/generated/models/MoveStructValue.ts:48]: what is?
everything under `generated` folder is generated by openAPI and therefore by whatever is written/added in the api level.

**Comment by 0xmaayan** [comment]
> Is this file: ecosystem/typescript/sdk_v2/src/internal/account.ts
> 
> the main one with the changes? seems like it has all of the `getData, getModules, getModule, getTransactions, getResources, getResource` functions
> 
> The rest is mostly generated types, right? Just making sure which part is actually new vs old generated stuff that we will phase out

yes this file, and also `/api/account.ts`. and also some changes in `api/aptos.ts` and `api/aptos_config.ts` - both are usage changes

File: ecosystem/typescript/sdk_v2/src/types/generated/models/MoveStructValue.ts (hunk 2 above)
Comment (xbtmatt) [reply ecosystem/typescript/sdk_v2/src/types/generated/models/MoveStructValue.ts:48]: ah yeah I figured, was just curious what the purpose of an empty type like this is

Comment (xbtmatt) [reply ecosystem/typescript/sdk_v2/src/types/generated/models/MoveStructValue.ts:48]: Ah ok, yea I get that you were just using the generated stuff, was just curious if there was some weird trick being used

**Review by gregnazario — APPROVED** [review]

File: ecosystem/typescript/sdk_v2/src/api/account.ts [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+   *
+   * @example An example of the returned account
+   * ```
+   * {
+   *    sequence_number: "1",
+   *    authentication_key: "0x5307b5f4bc67829097a8ba9b43dba3b88261eeccd1f709d9bde240fc100fbb69"
+   * }
+   * ```
+   */
+  async getData(args: { accountAddress: HexInput }): Promise<AccountData> {
```
[the remark this answers is not in the document]
Comment (xbtmatt) [reply ecosystem/typescript/sdk_v2/src/api/account.ts:36]: It does seem a little nebulous what "data" would be, `getAccountData` is kinda redundant but much clearer imo

[the remark this answers is not in the document]
Comment (0xmaayan) [reply ecosystem/typescript/sdk_v2/src/api/account.ts:36]: How about 'getInfo'?

File: ecosystem/typescript/sdk_v2/src/api/aptos_config.ts [hunk 4]
```diff
@@ -1,23 +1,70 @@
...
+          throw new Error("Please provide a custom faucet url");
+        return NetworkToFaucetAPI[this.network];
+      case AptosApiType.INDEXER:
+        if (this.indexer !== undefined) return this.indexer;
+        if (this.network === Network.CUSTOM && this.indexer === undefined)
+          throw new Error("Please provide a custom indexer url");
+        return NetworkToIndexerAPI[this.network];
+      default:
+        throw Error(`apiType ${apiType} is not supported`);
+    }
```
[the remark this answers is not in the document]
Comment (0xmaayan) [reply ecosystem/typescript/sdk_v2/src/api/aptos_config.ts:68]: wdym?

File: ecosystem/typescript/sdk_v2/src/api/account.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (xbtmatt) [reply ecosystem/typescript/sdk_v2/src/api/account.ts:36]: lol yeah dunno why that didn't come up in my mind! Perfect

**Comment by 0xjinn** [comment]
Curious - how are those model files get generated by OpenAPI? Is there any openAPI command you want to generate those models?

**Comment by 0xmaayan** [comment]
> Curious - how are those model files get generated by OpenAPI? Is there any openAPI command you want to generate those models?

we are stepping away from openAPI, we already stopped using the client in sdk_v1 and in v_2 we also stop using response types. check out https://github.com/aptos-labs/aptos-core/pull/9743

**Pull request merged**